Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -439,6 +439,61 @@ public IEnumerator TestDestroy()
             Assert.AreEqual(propBlock.GetColor("_Color"), FocusColor);
         }
 
+        [UnityTest]
+        /// <summary>
+        /// Tests button depth and focus state after enabling, disabling and re-enabling Interactable 
+        /// internally via SetDisable(true). The focus state after re-enabling should be false and button
+        /// depth should be in its default position.  This test is specifically addressing behavior described 
+        /// in issue 4967.
+        /// </summary>
+        public IEnumerator TestFocusStateOnDisableEnableInteractable()
+        {
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            int numSteps = 20;
+            Vector3 p1 = new Vector3(0.0f, 0f, 0f);
+            Vector3 p2 = new Vector3(0.05f, 0f, 0.51f);
+            Vector3 p3 = new Vector3(0.0f, 0f, 0.0f);
+
+            // Load the Model_PushButton interactable prefab
+            Interactable interactable;
+            Transform innerCylinderTransform;
+
+            InstantiatePressButtonPrefab(
+                new Vector3(0.0f, 0.0f, 0.5f),
+                new Vector3(-90f, 0f, 0f),
+                out interactable,
+                out innerCylinderTransform);
+
+            // OnClick, disable Interactable which sets Enabled property to false
+            interactable.OnClick.AddListener(() => { interactable.SetDisabled(true); });
+
+            // Get start position of the inner cylinder before button is pressed
+            Vector3 innerCylinderStartPosition = innerCylinderTransform.localPosition;
+
+            // Move the hand forward to press button
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService);
+            yield return PlayModeTestUtilities.MoveHandFromTo(p1, p2, numSteps, ArticulatedHandPose.GestureId.Poke, Handedness.Right, inputSimulationService);
+
+            // Ensure that the inner cylinder in the button has moved on press
+            yield return CheckButtonTranslation(innerCylinderStartPosition, innerCylinderTransform);
+
+            // Move the hand back
+            yield return PlayModeTestUtilities.MoveHandFromTo(p2, p3, numSteps, ArticulatedHandPose.GestureId.Poke, Handedness.Right, inputSimulationService);
+            yield return PlayModeTestUtilities.HideHand(Handedness.Right, inputSimulationService);
+            yield return new WaitForSeconds(ButtonReleaseAnimationDelay);
+
+            // Re-enable Interactable
+            interactable.SetDisabled(false);
+
+            // Make sure the button depth is back at the starting position when re-enable the gameObject
+            Assert.True(innerCylinderTransform.localPosition == innerCylinderStartPosition);
```

Comment:
Will this run instantly, or do you need to wait a few frames?

Reply:
This will run instantly, I added in a yield return null.